pkg/server/endpoints/registration/handler.go

```diff
@@ -545,6 +547,95 @@ func (h *Handler) ListAgents(ctx context.Context, listReq *registration.ListAgen
 	return &registration.ListAgentsResponse{Nodes: resp.Nodes}, nil
 }
 
+func (h *Handler) MintX509SVID(ctx context.Context, req *registration.MintX509SVIDRequest) (_ *registration.MintX509SVIDResponse, err error) {
+	counter := telemetry_registrationapi.StartMintX509SVIDCall(h.Metrics)
+	addCallerIDLabel(ctx, counter)
+	defer counter.Done(&err)
+
+	if req.SpiffeId == "" {
+		return nil, status.Error(codes.InvalidArgument, "request missing SPIFFE ID")
+	}
+	if len(req.Csr) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "request missing CSR")
+	}
+
+	spiffeID, err := idutil.NormalizeSpiffeID(req.SpiffeId, idutil.AllowTrustDomainWorkload(h.TrustDomain.Host))
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
```

Comment:
i think the nil check for `req.SpiffeId` and its normalization could probably be refactored into a single method that is called from both x509 and jwt implementations

Reply:
sure. done.